src/metrics/mongoose_metrics.erl

```diff
@@ -182,24 +184,47 @@ get_mnesia_running_db_nodes_count() ->
     {value, length(mnesia:system_info(running_db_nodes))}.
 
 remove_host_type_metrics(HostType) ->
-    HostTypeName = make_host_type_name(HostType),
+    HostTypeName = get_host_type_prefix(HostType),
     lists:foreach(fun remove_metric/1, exometer:find_entries([HostTypeName])).
 
 remove_all_metrics() ->
+    persistent_term:erase(?PREFIXES),
     lists:foreach(fun remove_metric/1, exometer:find_entries([])).
 
 %% ---------------------------------------------------------------------
 %% Internal functions
 %% ---------------------------------------------------------------------
 
+prepare_prefixes() ->
+    Prebuilt = maps:from_list([begin
+                                   Prefix = make_host_type_prefix(HT),
+                                   {Prefix, Prefix}
+                               end || HT <- ?ALL_HOST_TYPES ]),
+    Prefixes = maps:from_list([ {HT, make_host_type_prefix(HT)}
+                                || HT <- ?ALL_HOST_TYPES ]),
+    persistent_term:put(?PREFIXES, maps:merge(Prebuilt, Prefixes)).
+
 -spec all_metrics_are_global() -> boolean().
 all_metrics_are_global() ->
     mongoose_config:get_opt(all_metrics_are_global).
 
+get_host_type_prefix(HostType) when is_atom(HostType) ->
```

Comment:
Is this atom always `global`? If so, maybe we could match on it explicitly and add a type spec with `mongooseim:host_type_or_global()`? It would help with type checking and understanding the code.

Reply:
Good point, I think it is always `global` indeed, will double-check and try to fix 👍🏽